src/reducers/fileTree.tsx

```diff
@@ -191,6 +190,100 @@ export const getRelativePath = ({
   return pathList[newIndex];
 };
 
+const findRelativeMessageUid = (
+  currentPath: string,
+  messageMap: LinterMessageMap,
+  pathList: string[],
+  position: RelativePathPosition,
+): LinterMessage['uid'] | null => {
+  let path = currentPath;
+  for (let i = 0; i < pathList.length; i++) {
+    const nextPath = getRelativePath({
+      currentPath: path,
+      pathList,
+      position,
+    });
+    const nextMessages = messageMap[nextPath];
+    if (nextMessages) {
+      const nextMessagesForPath = getMessagesForPath(nextMessages);
+      if (position === RelativePathPosition.previous) {
```

Comment:
I found this part hard to read because of the nested if branching and the verbose variable names. Maybe after improving test coverage you can shorten it? Something like this:

```diff
diff --git a/src/reducers/fileTree.tsx b/src/reducers/fileTree.tsx
index c956ffd..d20dc6a 100644
--- a/src/reducers/fileTree.tsx
+++ b/src/reducers/fileTree.tsx
@@ -205,14 +205,11 @@ const findRelativeMessageUid = (
     });
     const nextMessages = messageMap[nextPath];
     if (nextMessages) {
-      const nextMessagesForPath = getMessagesForPath(nextMessages);
-      if (position === RelativePathPosition.previous) {
-        const lastMessageIndex = nextMessagesForPath.length - 1;
-        if (lastMessageIndex >= 0) {
-          return nextMessagesForPath[lastMessageIndex].uid;
-        }
-      } else if (nextMessagesForPath.length) {
-        return nextMessagesForPath[0].uid;
+      const nextForPath = getMessagesForPath(nextMessages);
+      const nextIndex =
+        position === RelativePathPosition.previous ? nextForPath.length - 1 : 0;
+      if (nextForPath[nextIndex]) {
+        return nextForPath[nextIndex].uid;
       }
     }
     path = nextPath;
```